tests/system/python/api/test_common.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test Common (ping, shutdown, restart) REST API """
+
+import socket
+import subprocess
+import http.client
+import time
+import json
+import pytest
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+@pytest.fixture
+def get_machine_detail():
+    host_name = socket.gethostname()
+    # all addresses for the host
+    all_ip_addresses_cmd_res = subprocess.run(['hostname', '-I'], stdout=subprocess.PIPE)
+    ip_addresses = all_ip_addresses_cmd_res.stdout.decode('utf-8').replace("\n", "").strip().split(" ")
+    return host_name, ip_addresses
+
+
+class TestCommon:
+
+    def test_ping_default(self, reset_and_start_foglamp, foglamp_url):
+        conn = http.client.HTTPConnection(foglamp_url)
+        conn.request("GET", '/foglamp/ping')
+        r = conn.getresponse()
+        assert 200 == r.status
+        r = r.read().decode()
+        jdoc = json.loads(r)
+        assert len(jdoc), "No data found"
+        host_name, ip_addresses = get_machine_detail()
+        assert 1 < jdoc['uptime']
+        assert isinstance(jdoc['uptime'], int)
+        assert 0 == jdoc['dataRead']
+        assert 0 == jdoc['dataSent']
+        assert 0 == jdoc['dataPurged']
+        assert 'FogLAMP' == jdoc['serviceName']
+        assert host_name == jdoc['hostName']
+        assert ip_addresses == jdoc['ipAddresses']
+        assert 'green' == jdoc['health']
+        assert jdoc['authenticationOptional'] is True
+        assert jdoc['safeMode'] is False
+
+    @pytest.mark.skip(reason='Not Implemented yet')
+    def test_ping_when_auth_mandatory(self):
+        # TODO: Its a bit tricky to automate user input on shell when foglamp start
```

Comment:
can not we set config auth mandatory and call  `conn.request("PUT", '/foglamp/restart')` and verify: allowPing True and False cases.

Reply:
Nope its 403 Forbidden
And syslog says `WARNING: common: foglamp.services.core.api.common: Permission denied for Ping when Auth is mandatory.`